### PR TITLE
Add support for PHP 7 with default php.ini (short_open_tag=Off)

### PR DESCRIPTION
**Pull request by tditlu**
- Replaced all "`<?`" with "`<?php`" except "`<?=`" which is always available since PHP 5.4.0. _(Default behavior for PHP > 7.0.0)_
- Replaced `each` function with a `for` loop. _(Deprecated since PHP 7.2.0)_
- Silence `file_get_contents`, if file doesn't exists
- Replace `define`/`defined` to take string as argument
- Silence `rmdir` & add default values for `regtime`, `regip` & `orgacomment` for lipsum plugin
- Add `.gitignore` for generated files

### CODE REVIEW

File: www_admin/config.php [hunk 1]
```diff
@@ -241,7 +241,7 @@ function perform(&$msg) {
   return 1;
 }
 
-if ($_POST["main_www_dir"]) {
+if (isset($_POST["main_www_dir"])) {
```
Comment (Gargaj) [inline www_admin/config.php]: All lack of `isset` is deliberate, because it also tests for null/emptystring/etc.

File: www_admin/sqllib.inc.php [hunk 2]
```diff
@@ -172,12 +172,10 @@ static function UpdateRow($table,$o,$where)
...
     foreach($a as $k=>$v) {
-      if ($v===NULL)
-      {
+      if ($v===NULL) {
         $set[] = sprintf("`%s`=null",mysqli_real_escape_string(SQLLib::$link,$k));
-      }
-      else
-      {
+      } else {
+        if (is_bool($v)) $v = $v ? 1 : 0;
```
Comment (Gargaj) [inline www_admin/sqllib.inc.php]: That's a presumption that every bool value in SQL is sent to a numeric field.

File: www_admin/config.php (hunk 1 above)
Comment (tditlu) [reply www_admin/config.php]: Ok, then maybe I should change them to the `empty` function. (http://php.net/manual/en/function.empty.php).
Why i added `isset` is because i got notices in PHP 7 that `$_POST["main_www_dir"]` was not set.

File: www_admin/sqllib.inc.php (hunk 2 above)
Comment (tditlu) [reply www_admin/sqllib.inc.php]: Hmm, yes, the problem i detected here was that it was inserting `user.visible` it inserted it as an empty string '', my mysql version (5.7) then throwed an error, because `user.visible` is defined as a `TINYINT`, maybe it's better to solve the `$userdata` then...

File: www_admin/config.php (hunk 1 above)
Comment (Gargaj) [reply www_admin/config.php]: Or use a @; I personally don't care much for notices.

Comment (tditlu) [reply www_admin/config.php]: @ doesn't hide notices for unset variables, but adding `ini_set('error_reporting'` will solve it
